### DISCUSSION TITLE
How to minimize code to only support 1 mode? (YAML)

### DISCUSSION THREAD

**Discussion by gatopeich** [ajaxorg/ace#5282] [Q&A]
Trying to minimize ACE for use as a YAML viewer, I find no way to get rid of `mode-‎lua.js‎` and `worker-‎lua.js‎`.

Are they hardcoded somewhere, or am I missing something?

**Comment by nightwing** [comment]
Could you show your code, yaml mode does not load the lua mode, so if you import yaml directly without mode resolver, lua should  not be loaded.

**Reply by gatopeich** [reply]
This is my `index.js` for webpack:
```js
import ace from 'ace-builds/src-noconflict/ace';
import modeYaml from 'ace-builds/src-noconflict/mode-yaml';
ace.config.setModuleUrl('ace/mode/yaml', modeYaml);

export {
    ace
};
```

And this `webpack.config.js`:
```js
const path = require('path');
const CopyPlugin = require('copy-webpack-plugin');

module.exports = {
    mode: 'development',
    entry: './src/index.js',
    output: {
        filename: 'dependencies.js',
        path: path.resolve('../MyProject/www', ''),
        library: 'MyLibraries',
        libraryTarget: 'umd',
        globalObject: 'this',
    },
    resolve: {
        extensions: ['.js']
    },
    plugins: [
        new CopyPlugin({
            patterns: [
                {
                    from: 'node_modules/ace-builds/src-noconflict/',
                    to: path.resolve('../MyProject/www/ace-modules', ''),
                    filter: resourcePath =>
                        /worker-yaml\.js$/.test(resourcePath) ||
                        /worker-lua\.js$/.test(resourcePath) ||  // If I remove this, there are exceptions in Chrome console
                        /mode-lua\.js$/.test(resourcePath)     // the same here
                }
            ],
        })
    ]
};
```

**Reply by nightwing** [reply]
The issue must be due to `ace.config.setModuleUrl('ace/mode/yaml', modeYaml);`, since setModuleUrl expects a url not the js object.

You don't need to configure moduleUrl for yaml mode itself, but for worker you need something like `ace.config.setModuleUrl("ace/mode/yaml_worker", "ace-modules/worker-yaml.js");`
see https://github.com/ajaxorg/ace-samples/blob/old-worker/samples/ace-builds-webpack/src/index.js
